### ISSUE TITLE
Key usage flags confusion

### ISSUE THREAD

**Issue by dschuermann** [open-keychain/open-keychain#1009]
No time currently to investigate, but I saw something suspicious for key EBC4 A367 32D6 CB2C DF92 F38E C4CB FEE8 9B6D 8A3A

Enigmail displays the master key with all key flags, but it has been generated with OpenKeychain and indeed we show only "certify" for the master key

**Comment by Valodim** [comment]
Well, shit :hurtrealbad:

So there's actually three sort-of separate issues here:

* The owner of that key certified his own key, which dropped all preferences and key flags. This was brought up and fixed in #1016, let's hope not too many people screwed up their keys with that one...
* Revoked user ids were not properly shown in the last place if they were also the only primary key. Fixed in e71bd3d9dd80d7f9721192238b9f30093c1c6bfc
* We used the logical or'ed combination of key flags from all self-signatures in a key, not the flags of the most recent certificate. That's just plain wrong, I don't know why I ever implemented it that way. Fixed in a65edcdb2ffb7399daf5359f0f57bf791c23b7d8
